File: app/src/main/java/com/whoguri/learnmvp/javaBase/BaseActivityListener.java
package com.whoguri.learnmvp.javaBase;

// whoGuri 3/11/18

import android.content.Intent;
import android.support.v4.app.ActivityCompat;
import android.support.v4.app.Fragment;

public interface BaseActivityListener {
    void replaceFragment(Fragment fragment, boolean isAddToBackStack);
    void goToActivity(Intent intent,boolean isToFinish);
    void showToast(String message);
    void showProgress();
    void hideProgress();

}

File: app/src/main/java/com/whoguri/learnmvp/kotlinBase/BaseActivityListener.kt
package com.whoguri.learnmvp.kotlinBase

// whoGuri 3/11/18

import android.content.Intent
import android.support.v4.app.Fragment

interface BaseActivityListener {
    fun replaceFragment(fragment: Fragment, isAddToBackStack: Boolean)
    fun goToActivity(intent: Intent, isToFinish: Boolean)
    fun showToast(message: String)
    fun showProgress()
    fun hideProgress()

}
